Volto docs integration

tiberiuichim opened the pull request:
@stevepiercy A bit late, but here it is. 

I've also added a Volto PR to match it. We're using that branch right now in the submodule, but that can be configured.

I've also played a bit with the Makefile, to try to make it more "plug and play", for example to avoid the virtualenv activation step.

Let me know if there's issues and I'll try to fix. Cheers!

tiberiuichim committed:
Add volto docs submodule

tiberiuichim committed:
Depend on submodules/volto for html, livehtml

tiberiuichim committed:
Set branch for submodule

tiberiuichim committed:
Remove unused target

tiberiuichim committed:
Add deps target, to be used by other targets

tiberiuichim committed:
Cleanup Makefile

mister-roboto commented:
@tiberiuichim thanks for creating this Pull Request and helping to improve Plone!

TL;DR: Finish pushing changes, pass all other checks, then paste a comment:
```
@jenkins-plone-org please run jobs
```

To ensure that these changes do not break other parts of Plone, the Plone test suite matrix needs to pass, but it takes 30-60 min.  Other CI checks are usually much faster and the Plone Jenkins resources are limited, so when done pushing changes and all other checks pass either [start all Jenkins PR jobs yourself](https://jenkinsploneorg.readthedocs.io/en/latest/run-pull-request-jobs.html#run-a-pull-request-job), or simply add the comment above in this PR to start all the jobs automatically.

Happy hacking!

tiberiuichim committed:
Better submodule init

tiberiuichim committed:
Better submodule init

tiberiuichim committed:
Use https git protocol

tiberiuichim committed:
Try to fix build

tiberiuichim committed:
Scratch that, we can't have nice things

sneridagh committed:
Improve Makefile and TOC

sneridagh committed:
Update submodule

stevepiercy commented:
w00t! It works! This is great!

We need to make a few docs updates to https://github.com/plone/documentation/edit/6-dev/docs/contributing/setup-build.md, and also update https://github.com/plone/documentation/blob/b37d36d7d7253fa6aea5bdd9c1641e1d3a1141de/.github/workflows/build_deploy.yml. I would also like @ksuess to give it a try.

sneridagh commented:
I gave it a try and LGTM!

Should we change from mkdocs to a MyST build in Volto as well? What do you think? I'd say so, to maintain coherence before all is settled. During the transition, it would be to maintain docs.voltocms.com up and running using the MyST build.

We will need to fix links and spelling, to keep the tests happy, right? I'm happy to help too.

tkimnguyen commented:
Point me at whatever you want me to review or tweak or edit!

tiberiuichim commented:
@tkimnguyen If you get this branch built, it uses https://github.com/plone/volto/pull/3001 as the linked "docs source", so any changes to Volto docs need to be done in the docs/volto subfolder and committed and pushed from the submodules/volto subfolder. When you make changes to the volto submodule, also make a commit in the main folder, to update the linked commit pointer for the submodule.

tiberiuichim commented:
@sneridagh It makes sense to convert volto docs to MyST, to keep things smooth.

stevepiercy commented:
If the docs are in CommonMark flavor of Markdown, there is no need to convert. MyST is a superset of CommonMark. If they have extensions, those parts are not supported and will need to be converted manually.

Also if you want to add reST or Sphinx features, such as cross-reference links, Intersphinx, admonitions, Glossary, and so on, then you can start adding them to the Volto docs.

stevepiercy commented:
We also need to add instructions for contributors to go to the source repo to edit volto docs, because they will expect to find them here. Especially the exact commands involved with submodules. I can never remember the `git submodule` commands.

There are also issues with the GitHub buttons on the deployed site with two repos. We'll continue to point to this repo, but we'll need to transfer Volto docs issues to the Volto repo as they come in. Oh, and I need to fix the branch it currently points to.

stevepiercy commented on .gitignore:
```diff
@@ -8,3 +8,8 @@
 pyvenv.cfg
 /_build
+*~
+lib64
```
I do not see where either of these are generated. Either please explain or put into `~/.gitignore_global` and remove.

tiberiuichim replied:
I have them in my .gitignore_global, for some reason they get ignored. I will remove them.

stevepiercy replied:
Fixed in 48b5ab20019214715481ba3e7f3dbc5f27affe22

stevepiercy commented on Makefile:
```diff
@@ -1,9 +1,12 @@
 # Makefile for Sphinx documentation
-#
+
+SHELL=bash
```
Is this required? I use zsh.

tiberiuichim replied:
I'll remove it, but even if you use zsh (I use fish), it's only for the variable handling internal to the processing of this Makefile. See https://www.gnu.org/software/make/manual/html_node/Choosing-the-Shell.html. I've added it thinking that it may be needed for the realpath part.

stevepiercy replied:
Thank you for the reference. After reading that link, I think keeping the `SHELL` var is a good idea.

stevepiercy replied:
This was resolved in e29c54136c1e9d75811c7a3fb2b344f10b635eaf

stevepiercy commented on Makefile:
```diff
@@ -1,9 +1,12 @@
-#
+
+SHELL=bash
+.DEFAULT_GOAL = all
```
Please remove. This prevents using just `make` on the command line to output the list of available make targets and their comments.

tiberiuichim replied:
Ok, I will set "help" as the default goal. The problem is that makefiles use the first defined target when you simply run `make`, so it can be confusing if people aren't familiar with makefiles.

stevepiercy replied:
I think you are right. `make help` is more understandable and would be a good default target for just `make`.

We will also need to update the Contributing docs to use `make help` instead of just `make` to get the list of targets and their comments. Explicit is better than implicit.

tiberiuichim replied:
In the latest version of the Makefile "help" is now the default target. I think that's the best. Usually "all" is the default target in makefile, when I've set that it was by force of habit.

stevepiercy replied:
Fixed in 3ee76c9f93654dbbe9505b70c6054ccf9ca30797

stevepiercy commented on Makefile:
```diff
@@ -27,57 +30,68 @@ clean: ## Clean build directory
 
 .PHONY: build
-build:  ## Set up training: Install requirements
+build:		## Set up training: Install requirements
```
Please revert replacement of spaces with tabs for all comment indents.

tiberiuichim replied:
Makefile targets list their dependencies after the colon, without some spacing it might look confusing, so I typically use two tabs to separate the help comment from the rest of the target definition. I will remove the tabs.

stevepiercy replied:
Thank you. Two spaces is good enough, is familiar to Python inline comments after a statement, and what we've already done with Trainings.

stevepiercy replied:
This was fixed in a couple of commits 042ad3327316f4a3a7932eb489da4ebf515a5e6b and 024c8fee982632e0998dce17d70a91bf50d53e2b

stevepiercy commented on Makefile:
```diff
@@ -27,57 +30,68 @@ clean: ## Clean build directory
+	bin/pip install -r requirements.txt
+
+docs/volto:
+	# git submodule add git@github.com:plone/volto.git submodules/volto
```
What is the purpose of this comment?

tiberiuichim replied:
I can remove. Just as a reminder of how the submodule was added.

stevepiercy replied:
That is what I assumed. Let's get it documented in `docs/contributing/authors.md`, as it is one of those things that is hard to remember, and we might need to do this for other repos (maybe restapi?). I've added this as a task

stevepiercy replied:
@tiberiuichim please check the commit 649811f0d0ddabf7f918e234c6d76e5fc37415a0 to make sure I understood how to do this.

stevepiercy replied:
Resolving per subsequent reviews from @ksuess and @nileshgulia1

stevepiercy commented on Makefile:
```diff
@@ -188,10 +202,14 @@ test: clean linkcheck spellcheck  ## Run linkcheck, spellcheck
+
+.PHONY: serve-docs
+serve-docs: html		## Start an HTTP server for docs
+	python -m http.server --directory ./_build/html
```
I think this target is redundant to `livehtml`. I don't think we need this, unless I am missing something.

tiberiuichim replied:
I've added it as a way to test the generated static html files. It was unclear to me how Sphinx generates the toctree based navigation and why it missed, sometimes, the section that I was adding. I can remove it, but I find it reassuring to have an easy way to "test the final build".

stevepiercy replied:
livehtml does that and more.

Sometimes you will need to do a `make clean`, as Sphinx only updates files that have changed and not the navigation site-wide. I do that when I edit `toctree` as it is confusing.

There are some things that you need to be aware of, but if you follow the pattern established at each level of docs (`docs/index.md` and `docs/classic-ui/index.md` for example), then the navigation should be happy. Feel free to ping me on Discord.

stevepiercy replied:
Fixed in 3ee76c9f93654dbbe9505b70c6054ccf9ca30797

stevepiercy reviewed (changes requested):
I have requested a few changes and have some questions for your consideration. Thank you!

stevepiercy committed:
Align all the =

stevepiercy committed:
Merge branch 'volto_docs' of https://github.com/plone/documentation into volto_docs

stevepiercy committed:
Place the Makefile variables under their appropriate comments

stevepiercy commented on Makefile:
```diff
@@ -27,57 +29,68 @@ clean: ## Clean build directory
+bin/python:
+	python3 -m venv . || virtualenv --clear --python=python3 .; \
+	bin/python -m pip install --upgrade pip; \
+	bin/pip install -r requirements.txt
```
This target is the same as `build`, but with statement delimiters. These two targets should be merged into one. I prefer the cleaner syntax of `build` without statement delimiters, and prefer the target name of `bin/python`. Is there a reason for using statement delimiters?

stevepiercy replied:
Fixed in 3ee76c9f93654dbbe9505b70c6054ccf9ca30797

stevepiercy commented:
To keep this PR short, let's focus on integration, specifically how to pull in the Volto docs and build them in this repo, and how to make updates to them. Content and conversion to MyST (if needed) can be done in separate PRs.

Here's a list of remaining tasks, as far as I see. Please feel free to add or edit.

- [x] Resolve Review comments and questions.
- [x] Update instructions in `docs/contributing/setup-build.md`, per resolution of Review.
- [x] Update `docs/contributing/setup-build.md` to reflect `make help` instead of just `make`.
- [x] Add section for how to add a remote repo to the docs in `docs/contributing/authors.md`. See https://github.com/plone/documentation/pull/1160#discussion_r790799617
- [x] Add instructions of how to update Volto docs, then pull those updates into this repo, in `docs/contributing/index.md`.
- [x] Add instructions for Volto contributions under the section `Making contributions on GitHub` in `docs/contributing/index.md`.

stevepiercy committed:
Change to bulleted list, and attempt to update volto docs with Plone Documentation

stevepiercy commented:
> @tkimnguyen If you get this branch built, it uses [plone/volto#3001](https://github.com/plone/volto/pull/3001) as the linked "docs source", so any changes to Volto docs need to be done in the docs/volto subfolder and committed and pushed from the submodules/volto subfolder. When you make changes to the volto submodule, also make a commit in the main folder, to update the linked commit pointer for the submodule.

@tiberiuichim can you put this into commands? Also what do you consider to be the "main folder"? Finally are the changes supposed to be made in this repo or volto's?

FWIW, I edited a volto file in this repo, pushed the commit to this branch/PR, but it does not go upstream to the volto repo. We want a process where changes are pushed to the upstream volto repo. Do developers need to add a git remote?

Please let us know. Thank you!

tiberiuichim commented:
@stevepiercy Here's my workflow to edit the Volto docs:

```
cd docs/volto
edit + save any docs/volto/*.md
cd ../submodules/volto
git commit -am "Message intended for plone/volto repo"
git push
cd ../../
git commit submodules/volto -m "Updated submodule/volto tip"
git push
```

Basically, treat submodules/volto as a separate repo. Branch it, commit/push separately in it, etc. Then, once the plone/volto repo (let's say master, but for now is the plone6docs branch) has the documentation updates in it, make sure to update the local copy to that HEAD and then go to the plone/documentation folder and make a commit with the SHA of the module. I think the key to understanding is looking at this page, https://github.com/plone/documentation/tree/volto_docs/submodules notice that the volto link points to a particular commit. Basically, we need a separate commit inside "documentation" to update what the submodule commit tip should be.

At some point we will want to update the submodule, and that can be done with `git submodule update`.

tiberiuichim committed:
Remove tabs in target definitions; remove serve-docs; remove SHELL, add distclean target and refactor build target; refactor bin/python target

tiberiuichim committed:
Remove target html dependency on deps, to fix netlify build

tiberiuichim committed:
Try with bash shell

stevepiercy committed:
Align =
Format comments consistently and improve their description accuracy

stevepiercy committed:
Add ./pyvenv.cfg to distclean target
Remove build target as redundant

stevepiercy committed:
Actually clean `BUILDDIR` by removing it completely, not just stuff inside it

stevepiercy committed:
Fix help text comment
Add `deps` target to `html` target. We will not use Netlify, after we set up our own hosting infrastructure

stevepiercy committed:
Restore help message display. '##' is required

stevepiercy committed:
Set proper build dir for `manual` target

stevepiercy committed:
Update submodules/volto tip

ksuess commented:
I can build. 
I can modify a Volto doc [0] and see the change in the build with `make livehtml`.
I did not risked to make a submodule update to commit to https://github.com/plone/volto/. Is this already in place? I do not want to interfere with ongoing work. Can I test and / or help?

[0]

````
```{note}
The `Listing` block already supports several of them (only in the "template" or…
```
````

tiberiuichim commented:
@ksuess You can use the following workflow (in generic terms):

- Right now we're using the plone6-docs branch from plone/volto. This is linked as a submodule in plone/documentation.
- When you pull this plone/documentation branch, it will set the submodules/volto folder to a particular commit, which you can see here: https://github.com/plone/documentation/tree/volto_docs/submodules. From git's point of view, the submodules/volto is on a "detached head", as it doesn't track a particular branch. So you need to run:

```
cd submodules/volto
git checkout plone6-docs
```
Now you can edit the files in docs/volto, they're symlinked from submodules/volto. When you want to make a commit, go to the submodules/volto and just do a commit there, with:

```
git add changed/files
git commit -m "..."
```

Once that's done, you need to go also to the root folder of this repo, and do another commit there:

```
cd ../../
git add submodules/volto
git commit -m "Updated submodules/volto tip"
```

And that is all. If I'm a contributor and I pull from the repo, my subfolders/volto repo will be updated to the same tip that you have commited for submodules/volto.

I have discussed with Steve and, for the future (at least for the immediate future) we have the following principles:

- Volto documentation will continue to be updated in plone/volto (and @stevepiercy will try to get a Myst-based setup for plone/volto)
- Whenever there's a Volto release, we can have a process to update the tip of the submodules/volto folder to that release, so that we track Volto releases.
- There will be a bit of learning curve now, in the beginning, while we adjust to the submodules workflow. In the future there will be less friction, I would say, as the Volto documentation is directly maintained by the Volto team, so only the Documentation team will not do a lot of work there (except at this stage, while we bring everything in line). Reusing the plone/documentation Sphinx stack in plone/volto will ensure that we keep things compatible and smooth running.

@sneridagh FYI ^

stevepiercy committed:
Update setup-build.md with correct instructions and descriptions of commands

stevepiercy committed:
First draft of contributing with Volto submodules.
Move existing "Managing contributions on GitHub" section after "Contributor Roles" section

stevepiercy committed:
Update submodules/volto tip

stevepiercy commented:
I added documentation for how to update volto docs with submodules in https://github.com/plone/documentation/pull/1160/files#diff-1aeecae8f389487f8c42feed5ae17f247adf21aea14e59918c3b6ac7f8d1b033

I also ran through that process, using https://github.com/plone/volto/pull/3017 as an example, and it seems to have worked for me.

I would appreciate someone else running through it and try to break it or improve it. @tkimnguyen @ksuess @tiberiuichim @sneridagh

stevepiercy committed:
First draft of contributing with Volto submodules.
Move existing "Managing contributions on GitHub" section after "Contributor Roles" section

stevepiercy committed:
Clean up .gitignore. We previously removed node stuff, and the other two should be in `~/.gitignore_global`

stevepiercy committed:
Add instructions for how to add external package documentation to Plone 6 Documentation

stevepiercy commented:
I completed my [above checklist](https://github.com/plone/documentation/pull/1160#issuecomment-1019925157), except for a review from @tiberiuichim in https://github.com/plone/documentation/pull/1160#discussion_r795311314

nileshgulia1 reviewed (approved):
I gave it a try, LGTM.

stevepiercy commented:
@nileshgulia1 thaks for the verification!

In the spirit of "merge first, ask questions later", keep momentum, and to unblock backend and classic-ui from contributing, I will merge this PR by end of day Tue, Feb 1. I would still like one more reviewer before merging. Always good to have another pair of eyes. Thank you!

stevepiercy committed:
Fix all broken links and redirects, except:
- volto/upgrade-guide/index.md:283: [broken] https://github.com/kitconcept/plone.volto/pull/29: 404 Client Error: Not Found for url: https://github.com/kitconcept/plone.volto/pull/29
- volto/addons/i18n.md:15: [broken] https://github.com/plone/volto/blob/master/src/i18n.js: 404 Client Error: Not Found for url: https://github.com/plone/volto/blob/master/src/i18n.js

stevepiercy committed:
Update submodules/volto tip

ksuess commented:
I am puzzled and need an update. If we review /plone/documentation/pull/1160 then /plone/volto/tree/plone6-docs is pulled in. is /plone/volto/tree/plone6-docs the branch we should look at (it seems to be the one that's adjusted to new Plone 6 docs, but not up to date with master as it was not freezed, and naturally not mergable anymore in Volto with no mkdocs artifacts)

ksuess commented:
I think http://127.0.0.1:8000/contributing/authors.html#importing-external-docs-with-submodules is not a topic for authors but for documentation administrators.

ksuess commented on docs/contributing/index.md:
```diff
@@ -53,35 +79,167 @@ Tap, click, or hover over the GitHub Octocat icon for options.
+
+(contributing-documentation-only-label)=
+
+### Working with only the `documentation` repository
```
```suggestion
### Working with the `plone/documentation` repository
```
- explicit mention of plone/documentation in contrary to plone/volto/docs
- request to native speakers: where and if to put the  word 'only'?

stevepiercy replied:
Fixed in e8d1338865d54b825addfe54761b10285d866628

The intent is to say that the contributor is working with only this repo and no other.

"Only" is a fascinating English word. It gets tossed around for emphasis and meaning. It's hard to explain. Here's a good example.

https://englishwithnab.com/place-the-word-only-anywhere-on-the-sentence/

ksuess commented on docs/contributing/index.md:
```diff
@@ -53,35 +79,167 @@ Tap, click, or hover over the GitHub Octocat icon for options.
+
+### Working with only the `documentation` repository
+
+This section describes how to make contributions to files in the `documentation` repository only, and excludes `volto` files.
```
```suggestion
This section describes how to make contributions to files in the `plone/documentation` repository only, and excludes `plone/volto/docs` files.
```

ksuess commented on docs/contributing/index.md:
```diff
@@ -53,35 +79,167 @@ Tap, click, or hover over the GitHub Octocat icon for options.
+
+This section describes how to make contributions to files in the `documentation` repository only, and excludes `volto` files.
+
+1.  From the project root directory, synch your local `6-dev` branch with its remote.
```
```suggestion
1.  From the project root directory, sync your local `6-dev` branch with its remote.
```

stevepiercy replied:
I am showing my age. Trends over the last two decades show that "sync" is now the more common usage. Reference: https://grammarhow.com/sync-vs-synch/

ksuess commented on docs/contributing/index.md:
````diff
@@ -53,35 +79,167 @@ Tap, click, or hover over the GitHub Octocat icon for options.
+    {ref}`setup-build-available-documentation-builds-label`.
+    ```
+
+1.  After the builds pass, commit changes to your branch, and push it to the remote repository on GitHub.
````
```suggestion
1.  After the build pass, commit changes to your branch, and push it to the remote repository on GitHub.
```

stevepiercy replied:
There are multiple builds that must pass, not just one.

ksuess commented on docs/contributing/index.md:
```diff
@@ -53,35 +79,167 @@ Tap, click, or hover over the GitHub Octocat icon for options.
+
+### Working with both the `documentation` and `volto` repositories
+
+When you want to edit files within the `docs/volto` subdirectory, the process is slightly different.
```
```suggestion
If you want to edit the Volto documentation: files within the `docs/volto` subdirectory, the process is slightly different.
```

stevepiercy replied:
There is an English phrase, "It's not 'if', but 'when`," meaning that something is inevitable. You're right, it is not inevitable, but it is from only my perspective 😆 .

Fixed in 32fb1ba43814949f76148cc9e6cafee726b8834a and 38c9510bd181045bee7592dec577fcf38e926ef4

ksuess commented on docs/contributing/index.md:
```diff
@@ -53,35 +79,167 @@ Tap, click, or hover over the GitHub Octocat icon for options.
+
+(contributing-documentation-and-volto-label)=
+
+### Working with both the `documentation` and `volto` repositories
```
```suggestion
### Editing Volto documentation

Working with both the `documentation` and `volto` repositories
```

stevepiercy replied:
Fixed in 89e81f9cddbd899687c0d9eaf2e6118963d076c4

ksuess commented on docs/contributing/index.md:
```diff
@@ -53,35 +79,167 @@ Tap, click, or hover over the GitHub Octocat icon for options.
+
+When you want to edit files within the `docs/volto` subdirectory, the process is slightly different.
+We use git submodules to manage multiple repositories.
+We imported the `volto` repository into the `documentation` repository as described in {doc}`setup-build`.
```
It is not described in the mentioned section.

stevepiercy replied:
@ksuess I rewrote this passage. Also see in `setup-build`:

> Then with a single command using `Makefile`, create a Python virtual environment, install project dependencies, **pull in Volto documentation as a git submodule**, build the docs, and view the results in a web browser by opening `/_build/html/index.html`.

ksuess commented on docs/contributing/index.md:
````diff
@@ -53,35 +79,167 @@ Tap, click, or hover over the GitHub Octocat icon for options.
+We imported the `volto` repository into the `documentation` repository as described in {doc}`setup-build`.
+
+```{important}
+We currently use the branches `plone/documentation@6-dev` and `plone/volto@plone6-docs` as the main branches for developing Plnoe 6 Documentation.
````
```suggestion
We currently use the branches `plone/documentation@6-dev` and `plone/volto@plone6-docs` as the main branches for developing Plone 6 Documentation.
```

ksuess commented on docs/contributing/index.md:
````diff
@@ -53,35 +79,167 @@ Tap, click, or hover over the GitHub Octocat icon for options.
+These branches may change as we get closer to a production release.
+```
+
+1.  From the project root directory, synch your local `6-dev` branch with its remote.
````
native speaker: isn't it sync?

stevepiercy replied:
Fixed in e45ae4f067067f407d894cd72efce49ed5b50f62

ksuess commented on docs/contributing/index.md:
````diff
@@ -53,35 +79,167 @@ Tap, click, or hover over the GitHub Octocat icon for options.
+    git push
+    ```
+
+1.  Now return to the project root directory, and update the submodule to point to the commit you just made, and push your changes to the remote repository.
````
```suggestion
1.  Now return to the project root directory, and update the submodule to point to the commit you just made, and push your changes to the remote /plone/documentation repository.
```

stevepiercy replied:
The remote branch could be in either a user's fork or the upstream repo `/plone/documentation`. Not being explicit allows either usage.

ksuess committed:
Update submodules/volto tip

ksuess commented on docs/contributing/index.md:
````diff
@@ -53,35 +79,167 @@ Tap, click, or hover over the GitHub Octocat icon for options.
+    git push
+    ```
+
+1.  Visit the GitHub `volto` repository, and [create a pull request](https://github.com/plone/volto/compare) against the branch `plone6-docs`.
````
This link is OK. But a link to a compare with the branch would be even better. Anyone expert on this field?

stevepiercy replied:
Yes, I tried, but it is a chicken-and-egg thing. You cannot create a link to compare a branch that does not exist. This is the best default option I could come up with.

Anyway, here is an example of comparing `master` as the base against `plone6-docs`: https://github.com/plone/volto/compare/master...plone6-docs

ksuess commented on docs/contributing/index.md:
````diff
@@ -53,35 +79,167 @@ Tap, click, or hover over the GitHub Octocat icon for options.
+    ```
+
+1.  Visit the GitHub `volto` repository, and [create a pull request](https://github.com/plone/volto/compare) against the branch `plone6-docs`.
+1.  Members who subscribe to the repository will receive a notification and review your request.
````
```suggestion
1.  Members who subscribe to the plone/volto repository will receive a notification and review your request.
```

stevepiercy replied:
Fixed in e06cdbb891c7417b07a076bd0d48c38724c0d160

ksuess commented on docs/contributing/index.md:
````diff
@@ -53,35 +79,167 @@ Tap, click, or hover over the GitHub Octocat icon for options.
+    make html
+    make linkcheck
+    ```
+
````
Add a changelog entry.

why not introduce  ## Documentation section in plone/volto?

```
### Documentation

- Add metadata on i18n page @janedoe
```

stevepiercy replied:
Agreed. I'll just move things around, and if anyone complains, slap me with a fish.

ksuess commented:
Still missing: 
- Merge of PR on plone/volto (if it affects plone/volto/docs)  triggers a rebuild of docs.plone.org

stevepiercy commented:
> I think http://127.0.0.1:8000/contributing/authors.html#importing-external-docs-with-submodules is not a topic for authors but for documentation administrators.

Yeah, this is not a great place for this content. What do you think of adding a page `contributing/admin.md` for this content and potential other future admin things?

stevepiercy committed:
Disambiguate the repository from volto/docs

stevepiercy committed:
Merge branch 'volto_docs' of github.com:plone/documentation into volto_docs

stevepiercy committed:
Update docs/contributing/index.md

Co-authored-by: Katja Süss <k.suess@rohberg.ch>

stevepiercy committed:
Update docs/contributing/index.md

Co-authored-by: Katja Süss <k.suess@rohberg.ch>

stevepiercy committed:
Disambiguate repos and location of files when editing Volto docs

stevepiercy committed:
If, not when

stevepiercy committed:
Condense heading and label

stevepiercy committed:
Update docs/contributing/index.md

Co-authored-by: Katja Süss <k.suess@rohberg.ch>

stevepiercy committed:
sync not synch

stevepiercy committed:
Merge branch 'volto_docs' of github.com:plone/documentation into volto_docs

ksuess commented:
> > I think http://127.0.0.1:8000/contributing/authors.html#importing-external-docs-with-submodules is not a topic for authors but for documentation administrators.
> 
> Yeah, this is not a great place for this content. What do you think of adding a page `contributing/admin.md` for this content and potential other future admin things?

yes, sounds fine.

stevepiercy committed:
Use explicit repo name plone/volto

stevepiercy committed:
Create new Admin Guide, moving content from Authors Guide

stevepiercy commented:
@ksuess I am done with the changes. This is ready for a final review. Thank you!

ksuess reviewed (approved):
Awesome! 
Maybe just integrating your new admistrators guide in toc/navigation.

polyester commented:
Did drive-by testing now on this branch, Ubuntu 20.04. It builds, live-html, etc. Looks good to me.

 \o/

stevepiercy committed:
Add admins to toctree

stevepiercy committed:
Update reference to correct label

stevepiercy commented:
> Awesome! Maybe just integrating your new admistrators guide in toc/navigation.

Thank you, and fixed.

Also found and squashed a couple of other little bugs.

stevepiercy committed:
Update reference to correct label, this time for sure!

stevepiercy committed:
Update submodules/volto tip

stevepiercy committed:
Attempt to get Netlify builds to pass

stevepiercy committed:
Attempt to get Netlify builds to pass - use sudo

stevepiercy committed:
Pin docutils for sphinx-book-theme

stevepiercy committed:
Attempt to get Netlify builds to pass using its system Python because we cannot `apt install python3.8-venv`

stevepiercy committed:
Attempt to get Netlify builds to pass - repeat myself

stevepiercy committed:
Attempt to get Netlify builds to pass

stevepiercy committed:
Update submodules/volto tip

stevepiercy commented:
Netlify preview builds work again.

The test failures can be ignored for now, as we are still working on infrastructure.

This is now ready for merge.

Let's see how it goes with https://github.com/plone/buildout.coredev/issues/768#issuecomment-1020578401
